File: lsp-check/sample.ts
interface User {
  id: number;
  name: string;
}

export function formatUser(user: User): string {
  return `${user.id}: ${user.name}`;
}

export function greetAll(users: User[]): string[] {
  return users.map((u) => formatUser(u));
}

const admin: User = { id: 1, name: "root" };
export const banner = formatUser(admin);

File: lsp-check/sample.py
from dataclasses import dataclass


@dataclass
class User:
    id: int
    name: str


def format_user(user: User) -> str:
    return f"{user.id}: {user.name}"


def greet_all(users: list[User]) -> list[int]:
    return [format_user(u) for u in users]


admin = User(id="1", name="root")
banner: int = format_user(admin)
